File: 4. Java/EX32.java
package Apostila;

import java.util.Locale;
import java.util.Scanner;

public class EX32
{
    public static void main(String[] args)
    {
        int n = 0;
        int limite = 10; // Limite da tabuada

        Locale.setDefault(Locale.US);

        try (Scanner ent = new Scanner(System.in))
        {
            // Solicita o valor da tabuada
            System.out.print("# Deseja a tabuada para qual valor: ");

            // Verifica se a entrada é um número inteiro
            if (ent.hasNextInt())
            {
                n = ent.nextInt();
            }
            else
            {
                System.out.println("# Valor inválido!");
                System.exit(0);
            }
            
            // Loop para gerar a tabuada
            System.out.println("");
            for (int i = 0; i <= limite; i++)
            {
                System.out.printf("# %d x %d = %d \n", n, i, n * i);
            }
            System.out.println("");
        }
    }
}

File: 6. Python/EX32.py
# Função para entrada de dados
def entDados():

    while True:
        try:

            # Solicita um número para a tabuada
            n = int(input("# Deseja a tabuada para qual valor: "))

            # Verifica se o número é positivo
            if n > 0:
                return n  # Retorna o número válido

            else:
                continue  # Repete se não for positivo

        except Exception:
            print("# Entrada Invalida !")  # Mensagem de erro para entrada inválida

# Função principal que imprime a tabuada
def main(n):

    for i in range(1, 11, 1):

        # Exibe a tabuada
        print(f"# {n} x {i} = {n * i}")

# Executa a função principal
if __name__ == '__main__':
    main(entDados())  # Chama a função de entrada e passa o resultado para main
